apps/etterlatte-saksbehandling-ui/client/src/Versioncheck.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react'
+import { hentClientConfig } from '~clientConfig'
+import { useAppSelector } from '~store/Store'
+import { Alert, BodyLong, Button, Modal } from '@navikt/ds-react'
+
+const Versioncheck = () => {
+  const appVersion = useAppSelector((state) => state.appReducer.appversion)
+  const [isOutdated, setIsOutdated] = useState<boolean>(false)
+  const [timeoutReloadPage, setTimeoutReloadPage] = useState<number | null>(null)
+
+  useEffect(() => {
+    const timeout5minutesinms = 300000
+    const tisekunderiMs = 10000
+    let intervalVersionCheck: null | number = null
+    let timeoutReload: null | number = null
+    intervalVersionCheck = window.setInterval(() => {
+      hentClientConfig().then((res) => {
+        if (res.status === 'ok') {
+          const fetchedAppversion = res.data.cachebuster
+          if (appVersion && fetchedAppversion !== appVersion) {
+            setIsOutdated(true)
+            const timeoutreftmp = window.setTimeout(() => {
+              window.location.reload()
+            }, tisekunderiMs)
+            setTimeoutReloadPage(timeoutreftmp)
+            timeoutReload = timeoutreftmp
+          }
+        }
+      })
+    }, timeout5minutesinms)
+
+    return () => {
+      timeoutReload && window.clearTimeout(timeoutReload)
+      intervalVersionCheck && window.clearTimeout(intervalVersionCheck)
+    }
+  }, [])
+
+  return (
+    <>
+      <Modal
+        header={{
+          label: 'Ny oppdatering',
+          heading: 'Utdatert web applikasjon',
+          closeButton: false,
+        }}
+        open={isOutdated}
+      >
+        <Modal.Body>
+          <Alert variant="warning">Web applikasjonen din er utdatert. Laster siden på nytt om 10 sekunder.</Alert>
```

Comment:
Forslag til tekst:

Det er kommet en ny versjon av Gjenny. Nettsida vil automatisk laste på nytt om 10 sekunder for å hente inn den nyeste versjonen.

Har du endringer du ønsker å lagre først? Trykk avbryt. (Obs! dette kan medføre at du mangler enkelte funksjoner og havner i feilsituasjoner!)

Reply:
https://github.com/navikt/pensjon-etterlatte-saksbehandling/pull/2088/commits/d1acf0e50c248be69419efa99b50d1617611e16d